presto-main/src/main/java/com/facebook/presto/block/BlockJsonSerde.java

```diff
@@ -53,10 +55,11 @@ public Serializer(BlockEncodingSerde blockEncodingSerde)
         public void serialize(Block block, JsonGenerator jsonGenerator, SerializerProvider serializerProvider)
                 throws IOException
         {
-            SliceOutput output = new DynamicSliceOutput(64);
+            //  Encoding name is length prefixed as are other block data encodings
+            SliceOutput output = new DynamicSliceOutput(toIntExact(block.getSizeInBytes() + block.getEncodingName().length() + (2 * Integer.BYTES)));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).